ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureStorageCompactor.java

```diff
@@ -56,11 +73,13 @@ public AzureStorageCompactor(AzureBlobDataAccessor azureBlobDataAccessor, Cosmos
     this.azureMetrics = azureMetrics;
     this.retentionPeriodMs = TimeUnit.DAYS.toMillis(cloudConfig.cloudDeletedBlobRetentionDays);
     this.queryLimit = cloudConfig.cloudBlobCompactionQueryLimit;
+    this.purgeLimit = cloudConfig.cloudCompactionPurgeLimit;
     this.queryBucketDays = cloudConfig.cloudCompactionQueryBucketDays;
     this.lookbackDays = cloudConfig.cloudCompactionLookbackDays;
-    // TODO: change this
-    compactionTimeLimitMs = TimeUnit.HOURS.toMillis(cloudConfig.cloudBlobCompactionIntervalHours);
     requestAgent = new CloudRequestAgent(cloudConfig, vcrMetrics);
+    emptyCheckpoints = new HashMap<>();
```

Comment:
since references to this map get returned via different method calls, consider wrapping it in `Collections.unmodifiableMap` It also seems like it could be static.

Reply:
ok